core/src/main/java/aima/core/search/api/BidirectionalProblem.java

```diff
@@ -0,0 +1,20 @@
+package aima.core.search.api;
+
+/**
+ * @author manthan
+ */
+public interface BidirectionalProblem<A, S> extends
+        // S initialState()
+        InitialStateFunction<S>,
+        // List<A> actions(S s)
+        ActionsFunction<A, S>,
+        // List<S> results(S s, A a)
+        ResultFunction<A, S>,
+        // boolean isGoalState(S s)
+        GoalTestPredicate<S>,
+        // double stepCost(S s, A a, S s')
+        StepCostFunction<A, S> {
+    Problem<A,S> getOriginalProblem();
```

Comment:
I thought I'd submitted a response as regards this earlier. I do not wish to go with the BidirectionalProblem API in this form or require any changes to the NodeFactory. Could you explain to me the difference between the results from call BidrectionalProblem.initialState() and BidirectionalProblem.getOriginalProblem().initialState()? Similarly, for the other overlapping methods. Basically, I only see BidirectionalProblem needing to be a container for the original and reverse versions of the problem, it itself does not need to be one. Let me know if I'm missing something here.

Thanks

Ciaran